test/parallel/test-stream-readable-async-iterators.js

```diff
@@ -0,0 +1,216 @@
+'use strict'
+
+const common = require('../common');
+const { Readable } = require('stream');
+const assert = require('assert');
+
+common.crashOnUnhandledRejection();
+
+async function tests () {
+  await (async function() {
+    console.log('read without for..await');
+    const max = 5;
+    let readed = 0;
+    let received = 0;
+    const readable = new Readable({
+      objectMode: true,
+      read() {}
+    });
+
+    const iter = readable[Symbol.asyncIterator]();
+    const values = [];
+    for (var i = 0; i < max; i++) {
+      values.push(iter.next());
+    }
+    Promise.all(values).then(common.mustCall((values) => {
```

Comment:
This test should probably push different values and check that they are in order. For example:

```js
 values.forEach(common.mustCall(
    (item, i) => assert.strictEqual(item.value, i), 5));
```
And pushing with:

```js
for(let i = 0; i < 5; i++) {
  readable.push(i);
}
```

Reply:
added